benchmarks/src/main/java/io/druid/benchmark/query/SearchBenchmark.java

```diff
@@ -0,0 +1,301 @@
+  {
+    QueryToolChest toolChest = factory.getToolchest();
+    QueryRunner<T> theRunner = new FinalizeResultsQueryRunner<>(
+        toolChest.mergeResults(toolChest.preMergeQueryDecoration(runner)),
+        toolChest
+    );
+
+    Sequence<T> queryResult = theRunner.run(query, Maps.<String, Object>newHashMap());
+    return Sequences.toList(queryResult, Lists.<T>newArrayList());
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  public void queryIncrementalIndex(Blackhole blackhole) throws Exception
+  {
+    QueryRunner<SearchHit> runner = QueryBenchmarkUtil.makeQueryRunner(
+        factory,
+        "incIndex",
+        new IncrementalIndexSegment(incIndex, "incIndex")
+    );
+
+    List<Result<SearchResultValue>> results = SearchBenchmark.runQuery(factory, runner, QUERY);
+    List<SearchHit> hits = results.get(0).getValue().getValue();
+    for (SearchHit hit : hits) {
+      blackhole.consume(hit);
+    }
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  public void queryQueryableIndex(Blackhole blackhole) throws Exception
+  {
+    final QueryRunner<Result<SearchResultValue>> runner = QueryBenchmarkUtil.makeQueryRunner(
+        factory,
+        "qIndex",
+        new QueryableIndexSegment("qIndex", qIndex)
+    );
+
+    List<Result<SearchResultValue>> results = SearchBenchmark.runQuery(factory, runner, QUERY);
+    List<SearchHit> hits = results.get(0).getValue().getValue();
+    for (SearchHit hit : hits) {
+      blackhole.consume(hit);
+    }
+  }
+
+  // Currently redundant, only merges one segment, and runQuery() already calls toolchest.mergeResults()
+  // Kept here for reference, if in the future a benchmark that merges results from multiple runners is added.
+  /*
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  public void mergeQueryableIndex(Blackhole blackhole) throws Exception
+  {
+    final QueryRunner<Result<SearchResultValue>> baseRunner = QueryBenchmarkUtil.makeQueryRunner(
+        factory,
+        "qIndex",
+        new QueryableIndexSegment("qIndex", qIndex)
+    );
+
+    QueryRunner mergedRunner = factory.getToolchest().mergeResults(
+        new QueryRunner<Result<SearchResultValue>>()
+        {
+          @Override
+          public Sequence<Result<SearchResultValue>> run(
+              Query<Result<SearchResultValue>> query, Map<String, Object> responseContext
+          )
+          {
+            return baseRunner.run(query, responseContext);
+          }
+        }
+    );
+
+    List<Result<SearchResultValue>> results = SearchBenchmark.runQuery(factory, mergedRunner, QUERY);
+    List<SearchHit> hits = results.get(0).getValue().getValue();
+    for (SearchHit hit : hits) {
+      blackhole.consume(hit);
+    }
+  }
+  */
+
+
+  public void profile() throws Exception
```

Comment:
I think I'd really like to see this stuff be more linked with the rest of the tests. We eventually want an automated performance testing system (#2823) and the more we can be consistent about test style, the better.